src/meta/src/manager/catalog/database.rs

```diff
@@ -101,11 +102,32 @@ impl DatabaseManager {
                 .map(|database| (database.id, database)),
         );
         let schemas = BTreeMap::from_iter(schemas.into_iter().map(|schema| (schema.id, schema)));
-        let sources = BTreeMap::from_iter(sources.into_iter().map(|source| {
+        let sources = BTreeMap::from_iter(sources.into_iter().map(|mut source| {
             // TODO(weili): wait for yezizp to refactor ref cnt
             if let Some(connection_id) = source.connection_id {
                 *relation_ref_count.entry(connection_id).or_default() += 1;
             }
+
+            // This block here is for maintaining backward compatibility with older source formats when `format_encode_options` is
```

Comment:
How about introducing a separate function in catalog manager to rewrite in-compatible sources? Then you can use `commit_meta` instead of writing kv directly with a low level api.